Remove string solver iteration limit

smowton opened the pull request:
This could prevent a solution from being reached whenever more than 100 iterations
were required to reach one, and since it returned error on failure its termination
could never yield any useful result. Therefore let's simply let the user or driver
program / platform choose when to give up rather than impose an arbitrary limit.

See also https://diffblue.atlassian.net/browse/TG-2338

peterschrammel reviewed (commented):
There's an unused variable: https://travis-ci.org/diffblue/cbmc/jobs/339475394#L501

martin-cs reviewed (commented):
Removing a hard coded limit -> a good thing.

Removing the option of limiting the refinement loop -> I'd say it was useful to have a limit as then you can potentially fall back to other approaches.  I know at least one major user does this with the --refine option.

smowton committed:
Remove string refinement iteration limit

This sets the string refinement iteration limit to zero, i.e. unlimited.
The mechanism for providing a limit is left in place, for use probably driven
by a command-line argument by the first user to need that facility.

smowton commented:
@martin-cs I've left the option in there, I'll leave hooking it up to a command-line argument to the first interested user.

smowton commented:
tg https://github.com/diffblue/test-gen/pull/1492

peterschrammel commented on src/solvers/refinement/string_refinement.cpp:
```diff
@@ -900,7 +900,9 @@ decision_proceduret::resultt string_refinementt::dec_solve()
-  while((loop_bound_--)>0)
+  // loop_bound_ == 0 means no limit; otherwise set max iteration bound:
+  std::size_t refinement_iterations = 0;
+  while(loop_bound_ == 0 || (refinement_iterations++) < loop_bound_)
```
0 for no limit is a bit weird. Why not use numeric_limits<size_t>::max? Or do you expect more iterations than that?

smowton commented:
@peterschrammel fair enough, changed as you recommend

peterschrammel reviewed (approved):
Lgtm, modulo Ci passing
